extensions/gitpod-web/src/extension.ts

```diff
@@ -539,12 +527,27 @@ export function registerPorts(context: GitpodExtensionContext): void {
 	context.subscriptions.push(treeView);
 
 	// register webview
-	let portViewProvider: GitpodPortViewProvider | undefined;
-	if (isPortsViewExperimentEnable) {
-		vscode.commands.executeCommand('setContext', 'gitpod.portsView.visible', true);
-		portViewProvider = new GitpodPortViewProvider(context);
-		context.subscriptions.push(vscode.window.registerWebviewViewProvider(GitpodPortViewProvider.viewType, portViewProvider, { webviewOptions: { retainContextWhenHidden: true } }));
-	}
+	const portViewProvider = new GitpodPortViewProvider(context);
+	context.subscriptions.push(vscode.window.registerWebviewViewProvider(GitpodPortViewProvider.viewType, portViewProvider, { webviewOptions: { retainContextWhenHidden: true } }));
+
+	// new Ports view - try it out
+	let hasShownTryItOut = false;
+	treeView.onDidChangeVisibility(async (e) => {
+		const config = vscode.workspace.getConfiguration('gitpod.experimental.portsView');
+		if (hasShownTryItOut || !e.visible || config.get<boolean>('enabled') || config.get<boolean>('neverPrompt') === true) {
+			return;
+		}
+		hasShownTryItOut = true;
+		const openAction = 'Open';
+		const neverAgain = 'Don\'t Show Again';
+		const action = await vscode.window.showInformationMessage('Do you want to try with our new experimental Ports view?', openAction, neverAgain);
+		if (action === openAction) {
+			config.update('enabled', true, true);
+			setTimeout(() => vscode.commands.executeCommand('gitpod.portsView.focus'), 2000);
```

Comment:
If we don't delay call focus, it won't work since Ports view is not setup (show in panel) by vscode.

If we don't focus on this view, user may don't know where is the new Ports view

@jeanp413 Any suggestion? since we can't listen to is Ports view setup 😞

Reply:
This seems to be working https://github.com/gitpod-io/openvscode-server/pull/384/commits/1dae9c28c818d3f7340263fb5e62807a521a0373

Reply:
It's working now, thanks JP!! 👍 , (but I'm going to remove this `Try it Out` notification)

Reply:
Well, there it goes my precious hours of debugging and testing :rofl:

Reply:
😂 You shared the knowledge JP!!